socorro/lib/transaction.py

```diff
@@ -81,24 +78,21 @@ def retry(connection_context, quit_check, fun, **kwargs):
     :returns: varies
 
     """
-    last_failure = None
-
-    for backoff_time in BACKOFF_TIMES:
+    for retry_round, backoff_time in enumerate(BACKOFF_TIMES):
+        is_last_round = retry_round == (len(BACKOFF_TIMES) - 1)
```

Comment:
I think I would pre-calculate `len(BACKOFF_TIMES) - 1` and stick that in a variable. Something like this:

```
max_attempts = len(BACKOFF_TIMES) - 1

for retry_round, backoff_time in enumerate(BACKOFF_TIMES):
    is_last_round = retry_round == max_attempts
    ...
```

But either way is fine and gets the same behavior.